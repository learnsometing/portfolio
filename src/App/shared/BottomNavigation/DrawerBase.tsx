import React, { ReactNode } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Drawer from '@material-ui/core/Drawer';
import Paper from '@material-ui/core/Paper';

const FullHeightDrawer = styled(Drawer)`
  height: 100vh;
`;

const DrawerBG = styled(Paper)`
  overflow-y: auto;
  background-color: ${(props): string => props.theme.bgLightened};
  min-height: 100vh;
  overflow-x: hidden;
`;

interface DrawerBaseProps {
  isOpen: boolean;
  children: ReactNode;
}

const DrawerBase: React.FC<DrawerBaseProps> = ({ isOpen, children }) => {
  return (
    <FullHeightDrawer
      anchor={'top'}
      open={isOpen}
      elevation={2}
      transitionDuration={400}
      PaperProps={{ component: DrawerBG }}
      variant={'persistent'}
    >
      {children}
    </FullHeightDrawer>
  );
};

DrawerBase.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
};

export default DrawerBase;
