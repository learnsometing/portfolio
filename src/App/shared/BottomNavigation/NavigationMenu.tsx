import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ListItem from '@material-ui/core/ListItem';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';

import { AnchoredLink } from '../Navigation';
import { MenuCloseButton } from '../CloseButton';

const MenuItem = styled(ListItem).attrs({
  button: undefined,
})`
  height: calc((100vh - 4px - (3rem + 4vh)) / 4);
  margin: 0;
`;

const Menu = styled(List)`
  width: 100vw;
  padding: 0;
  background-color: ${(props): string => props.theme.bgLightened};
`;

const MenuLink = styled(AnchoredLink)`
  display: flex;
  width: 100%;
  height: 100%;
  padding: 0;
  justify-content: center;
  align-items: center;
`;

const MenuLinkText = styled(Typography).attrs({
  variant: 'h3',
  component: 'span',
})`
  color: ${(props): string => props.theme.textEmphasis};
`;

interface NavigationDrawerProps {
  toggle: () => void;
}

const NavigationMenu: React.FC<NavigationDrawerProps> = ({ toggle }) => {
  return (
    <Menu>
      <MenuItem onClick={toggle}>
        <MenuLink to="/#header">
          <MenuLinkText>Home</MenuLinkText>
        </MenuLink>
      </MenuItem>
      <Divider component={'li'} />
      <MenuItem onClick={toggle}>
        <MenuLink to="/#about">
          <MenuLinkText>About</MenuLinkText>
        </MenuLink>
      </MenuItem>
      <Divider component={'li'} />
      <MenuItem onClick={toggle}>
        <MenuLink to="/portfolio">
          <MenuLinkText>Portfolio</MenuLinkText>
        </MenuLink>
      </MenuItem>
      <Divider component={'li'} />
      <MenuItem onClick={toggle}>
        <MenuLink to="#contact">
          <MenuLinkText>Contact</MenuLinkText>
        </MenuLink>
      </MenuItem>
      <Divider />
      <MenuCloseButton onClick={toggle} />
    </Menu>
  );
};

NavigationMenu.propTypes = {
  toggle: PropTypes.func.isRequired,
};

export default NavigationMenu;
