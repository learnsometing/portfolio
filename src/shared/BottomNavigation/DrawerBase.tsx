import React, { ReactNode } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Drawer from '@material-ui/core/Drawer';
import Paper from '@material-ui/core/Paper';

const FullHeightDrawer = styled(Drawer)`
  height: 100vh;
`;

const DrawerBG = styled(Paper)`
  overflow-y: auto;
  background-color: #fdfde8;
  min-height: 100vh;
`;

interface DrawerBaseProps {
  isOpen: boolean;
  children: ReactNode;
}

const DrawerBase: React.FC<DrawerBaseProps> = ({ isOpen, children }) => {
  return (
    <FullHeightDrawer
      anchor={'top'}
      open={isOpen}
      elevation={2}
      transitionDuration={400}
      PaperProps={{ component: DrawerBG }}
    >
      {children}
    </FullHeightDrawer>
  );
};

DrawerBase.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
};

export default DrawerBase;
