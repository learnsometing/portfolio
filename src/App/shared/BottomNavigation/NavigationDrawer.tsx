import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ListItem from '@material-ui/core/ListItem';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';

import DrawerBase from './DrawerBase';
import { AnchoredLink } from '../Navigation';
import { MenuCloseButton } from '../CloseButton';

const MenuItem = styled(ListItem).attrs({
  button: undefined,
})`
  height: calc((100vh - 4px - (3rem + 4vh)) / 4);
  margin: 0;
`;

const Menu = styled(List)`
  width: 100vw;
  padding: 0;
  background-color: ${(props): string => props.theme.bgLightened};
`;

const MenuLink = styled(AnchoredLink)`
  display: flex;
  width: 100%;
  height: 100%;
  padding: 0;
  justify-content: center;
  align-items: center;
`;

const MenuLinkText = styled(Typography).attrs({
  variant: 'h3',
  component: 'span',
})`
  color: ${(props): string => props.theme.textEmphasis};
`;

interface NavigationDrawerProps {
  isOpen: boolean;
  toggle: () => void;
}

const NavigationDrawer: React.FC<NavigationDrawerProps> = ({
  isOpen,
  toggle,
}) => {
  return (
    <DrawerBase isOpen={isOpen}>
      <Menu>
        <MenuItem onClick={toggle}>
          <MenuLink to="/#header">
            <MenuLinkText>Home</MenuLinkText>
          </MenuLink>
        </MenuItem>
        <Divider component={'li'} />
        <MenuItem onClick={toggle}>
          <MenuLink to="/#about">
            <MenuLinkText>About</MenuLinkText>
          </MenuLink>
        </MenuItem>
        <Divider component={'li'} />
        <MenuItem onClick={toggle}>
          <MenuLink to="/#portfolio">
            <MenuLinkText>Portfolio</MenuLinkText>
          </MenuLink>
        </MenuItem>
        <Divider component={'li'} />
        <MenuItem onClick={toggle}>
          <MenuLink to="/#contact">
            <MenuLinkText>Contact</MenuLinkText>
          </MenuLink>
        </MenuItem>
        <Divider />
        <MenuCloseButton onClick={toggle} />
      </Menu>
    </DrawerBase>
  );
};

NavigationDrawer.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
};

export default NavigationDrawer;
