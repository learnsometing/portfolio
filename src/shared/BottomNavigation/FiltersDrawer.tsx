import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import DrawerBase from './DrawerBase';
import FiltersForm from './FiltersForm';

import CloseButton from '../CloseButton';

const Header = styled(Grid)`
  border-bottom: 1px solid rgba(0, 0, 0, 0.12);
  padding: 0.5rem 1.5rem;
  position: sticky;
  top: 0;
  background-color: #fdfde8;
  z-index: 1;
`;

const FilterDrawerHeading = styled(Typography).attrs({
  variant: 'h2',
})``;

interface FiltersDrawerProps {
  isOpen: boolean;
  filters: string[][];
  closeFiltersDrawer: () => void;
}

const FiltersDrawer: React.FC<FiltersDrawerProps> = ({
  isOpen,
  closeFiltersDrawer,
  filters,
}) => {
  return (
    <DrawerBase isOpen={isOpen}>
      <Header container justify={'space-between'} alignItems={'center'}>
        <FilterDrawerHeading>Filters</FilterDrawerHeading>
        <CloseButton
          aria-label="Close Filter Drawer"
          onClick={closeFiltersDrawer}
        />
      </Header>
      <FiltersForm filters={filters} onSubmitCallback={closeFiltersDrawer} />
    </DrawerBase>
  );
};

FiltersDrawer.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  closeFiltersDrawer: PropTypes.func.isRequired,
  filters: PropTypes.arrayOf(
    PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
  ).isRequired,
};

export default FiltersDrawer;
