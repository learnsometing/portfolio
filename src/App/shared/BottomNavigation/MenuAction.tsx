import React, { useState } from 'react';
import styled from 'styled-components';

import { MdMenu } from 'react-icons/md';

// Material-UI
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import Typography from '@material-ui/core/Typography';

import NavigationDrawer from './NavigationDrawer';

const MenuIcon = styled(MdMenu)`
  height: ${(props): string => props.theme.bottomNavigationIconSize};
  width: ${(props): string => props.theme.bottomNavigationIconSize};
  color: ${(props): string => props.theme.textEmphasis};
`;

const MenuAction: React.FC = () => {
  const [isNavDrawerOpen, setIsNavDrawerOpen] = useState(false);
  const toggleNavDrawer = (): void => {
    setIsNavDrawerOpen(!isNavDrawerOpen);
  };

  return (
    <>
      <BottomNavigationAction
        label={
          <Typography variant={'h5'} component={'span'}>
            Menu
          </Typography>
        }
        showLabel
        icon={<MenuIcon />}
        onClick={toggleNavDrawer}
      />
      <NavigationDrawer isOpen={isNavDrawerOpen} toggle={toggleNavDrawer} />
    </>
  );
};

export default MenuAction;
