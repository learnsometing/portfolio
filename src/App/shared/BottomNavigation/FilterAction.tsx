import React, { useState } from 'react';
import styled from 'styled-components';
import { MdFilterList } from 'react-icons/md';
import PropTypes from 'prop-types';

// Material-UI
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import Typography from '@material-ui/core/Typography';

import FiltersDrawer from './FiltersDrawer';

interface Props {
  allProjectTags: string[][];
}

const FilterIcon = styled(MdFilterList)`
  height: ${(props): string => props.theme.bottomNavigationIconSize};
  width: ${(props): string => props.theme.bottomNavigationIconSize};
  color: ${(props): string => props.theme.textEmphasis};
`;

const FilterAction: React.FC<Props> = ({ allProjectTags }) => {
  const [isFiltersDrawerOpen, setIsFiltersDrawerOpen] = useState(false);
  const openFiltersDrawer = (): void => {
    setIsFiltersDrawerOpen(true);
  };
  const closeFiltersDrawer = (): void => {
    setIsFiltersDrawerOpen(false);
  };

  return (
    <>
      <BottomNavigationAction
        label={
          <Typography variant={'h5'} component={'span'}>
            Filter
          </Typography>
        }
        showLabel
        icon={<FilterIcon />}
        onClick={openFiltersDrawer}
      />
      <FiltersDrawer
        isOpen={isFiltersDrawerOpen}
        closeFiltersDrawer={closeFiltersDrawer}
        allProjectTags={allProjectTags}
      />
    </>
  );
};

FilterAction.propTypes = {
  allProjectTags: PropTypes.arrayOf(
    PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
  ).isRequired,
};

FilterAction.defaultProps = {
  allProjectTags: [],
};

export default FilterAction;
