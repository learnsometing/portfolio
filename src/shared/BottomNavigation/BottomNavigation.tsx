import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { IconContext } from 'react-icons';
import { MdFilterList, MdMenu } from 'react-icons/md';
import Paper from '@material-ui/core/Paper';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import Typography from '@material-ui/core/Typography';

import NavigationDrawer from './NavigationDrawer';
import FiltersDrawer from './FiltersDrawer';

interface NavbarProps {
  isFiltrationDisabled: boolean;
}

const Navbar = styled(Paper).attrs({
  elevation: 2,
  square: true,
})<NavbarProps>`
  position: fixed;
  bottom: 0;
  right: 0;
  height: 80px;
  width: 100%;
  z-index: 2;
  background-color: #fdfde8;
  @media only screen and (min-width: 1280px) {
    display: none;
  }
`;

const CondensedNavbar = styled(Navbar).attrs({
  elevation: 2,
  square: true,
})<NavbarProps>`
  width: 80px;
  border-radius: 50%;
  margin: 1rem;
`;

export const ActionLabel = styled(Typography).attrs({
  variant: 'h6',
  component: 'span',
})`
  color: #f95738;
`;

interface Props {
  allProjectTags?: string[][];
}

const iconStyles = { size: '32px', color: '#f95738' };

const filterIcon = (
  <IconContext.Provider value={iconStyles}>
    <MdFilterList />
  </IconContext.Provider>
);

const menuIcon = (
  <IconContext.Provider value={iconStyles}>
    <MdMenu />
  </IconContext.Provider>
);

const BottomNavigationBar: React.FC<Props> = ({ allProjectTags }) => {
  const [isNavDrawerOpen, setIsNavDrawerOpen] = useState(false);
  const [isFiltersDrawerOpen, setIsFiltersDrawerOpen] = useState(false);

  const toggleNavDrawer = (): void => {
    setIsNavDrawerOpen(!isNavDrawerOpen);
  };

  const openFiltersDrawer = (): void => {
    setIsFiltersDrawerOpen(true);
  };

  const closeFiltersDrawer = (): void => {
    setIsFiltersDrawerOpen(false);
  };

  return allProjectTags && allProjectTags.length ? (
    <BottomNavigation component={Navbar} showLabels>
      <BottomNavigationAction
        label={<ActionLabel>Filter</ActionLabel>}
        showLabel
        icon={filterIcon}
        onClick={openFiltersDrawer}
      />
      <FiltersDrawer
        isOpen={isFiltersDrawerOpen}
        closeFiltersDrawer={closeFiltersDrawer}
        allProjectTags={allProjectTags}
      />
      <BottomNavigationAction
        label={<ActionLabel>Menu</ActionLabel>}
        showLabel
        icon={menuIcon}
        onClick={toggleNavDrawer}
      />
      <NavigationDrawer isOpen={isNavDrawerOpen} toggle={toggleNavDrawer} />
    </BottomNavigation>
  ) : (
    <BottomNavigation component={CondensedNavbar} showLabels>
      <BottomNavigationAction
        label={<ActionLabel>Menu</ActionLabel>}
        showLabel
        icon={menuIcon}
        onClick={toggleNavDrawer}
      />
      <NavigationDrawer isOpen={isNavDrawerOpen} toggle={toggleNavDrawer} />
    </BottomNavigation>
  );
};

BottomNavigationBar.propTypes = {
  allProjectTags: PropTypes.arrayOf(
    PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
  ),
};

BottomNavigationBar.defaultProps = {
  allProjectTags: [],
};

export default BottomNavigationBar;
