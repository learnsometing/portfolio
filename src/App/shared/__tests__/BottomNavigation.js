import React from 'react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import scTheme from '../SCTheme';

import BottomNavigation from '../BottomNavigation/BottomNavigation';

describe('BottomNavigation', () => {
  it('should match the snapshot', () => {
    const tree = create(
      <ThemeProvider theme={scTheme}>
        <BottomNavigation />
      </ThemeProvider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
