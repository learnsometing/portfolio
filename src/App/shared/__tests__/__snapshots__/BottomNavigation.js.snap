// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavigation should match the snapshot 1`] = `
.c0 {
  position: -webkit-sticky;
  position: sticky;
  bottom: 0;
  right: 0;
  height: 80px;
  width: 100%;
  z-index: 2;
  background-color: #3E476C;
  position: fixed;
  width: 80px;
  border-radius: 50%;
  bottom: 1rem;
  right: 1rem;
}

@media only screen and (min-width:1280px) {
  .c0 {
    display: none;
  }
}

<div
  className="MuiPaper-root BottomNavigation__Navbar-sc-1b8in4y-0 c0 MuiBottomNavigation-root MuiPaper-elevation2"
>
  <button
    className="MuiButtonBase-root MuiBottomNavigationAction-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper"
    >
      <svg
        fill="currentColor"
        height="32px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#F9E43B",
          }
        }
        viewBox="0 0 24 24"
        width="32px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
      <span
        className="MuiBottomNavigationAction-label"
      >
        <span
          className="MuiTypography-root MuiTypography-h5"
        >
          Menu
        </span>
      </span>
    </span>
  </button>
</div>
`;
